controllers/glanceapi_controller.go

```diff
@@ -797,10 +830,81 @@ func (r *GlanceAPIReconciler) reconcileNormal(
 			err.Error()))
 		return ctrl.Result{}, err
 	}
+
 	// At this point the config is generated and the inputHash is computed
 	// we can mark the ServiceConfigReady as True and rollout the new pods
 	instance.Status.Conditions.MarkTrue(condition.ServiceConfigReadyCondition, condition.ServiceConfigReadyMessage)
 
+	//
+	// Handle Topology
+	//
+	var topology *topologyv1.Topology
```

Comment:
@stuggi @olliewalsh do you think we should still wrap this `Handle Topology` logic, from L841 to L907 in a dedicated function? The advantage would be, in other operators, that we can reuse the same, common function, across components (e.g. `manilaAPI`, `manilaScheduler`, `manilaShares` controller might call the same big block). This would help to ensure consistency.
I'm looking if the `EnsureDeleteTopologyRef` logic (we have two calls) can be simplified here and reduce it to a single call (we call it both when `.Spec.TopologyRef` is `nil` and when `!= nil`, so I think I still have to push an update to simplify this further.

Reply:
If I read it correct, it should be generic to all consumers? if the list of parameters we'd have to pass in is not super big, also we could use a struct type for this for passing in the details. this would allow us to extend it and not have to change the func definition. condition handling would be outside the func. so yes I think its worth looking into it.